pinot-connectors/pinot-flink-connector/src/main/java/org/apache/pinot/connector/flink/sink/PinotSinkFunction.java

```diff
@@ -76,13 +84,25 @@ public PinotSinkFunction(PinotGenericRowConverter<T> recordConverter, TableConfi
     _schema = schema;
     _segmentFlushMaxNumRecords = segmentFlushMaxNumRecords;
     _executorPoolSize = executorPoolSize;
+    _segmentNamePrefix = DEFAULT_UPLOADED_REALTIME_SEGMENT_PREFIX;
+  }
+
+  public PinotSinkFunction(PinotGenericRowConverter<T> recordConverter, TableConfig tableConfig, Schema schema,
+      long segmentFlushMaxNumRecords, int executorPoolSize, String segmentNamePrefix,
+      @Nullable Long segmentUploadTimeMs) {
+    this(recordConverter, tableConfig, schema, segmentFlushMaxNumRecords, executorPoolSize);
+    if (!segmentNamePrefix.isBlank()) {
```

Comment:
this means we enforce a non-empty segment name prefix? could there be a need to support no prefix?

Reply:
It would depend on the segmentNameGenerator property if the prefix is needed, even though configured. But rethinking this, there are two ways to configure this:
- In table config as batchIngestionConfig
- In code via PinotSinkFunction config -> SegmentGenerator config

Right now, latter will always take precedence with a default value. Refactoring so that default value is used when neither is configured otherwise